pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -159,7 +170,14 @@ private void start() {
                 return null;
             });
         }
+    }
 
+    private ProducerImpl<T> createProducer(final int partitionIndex) {
+        String partitionName = TopicName.get(topic).getPartition(partitionIndex).toString();
+        ProducerImpl<T> producer = client.newProducerImpl(partitionName, partitionIndex,
+                conf, schema, interceptors, new CompletableFuture<>());
+        producers.put(partitionIndex, producer);
```

Comment:
Isn't it necessary to add an entry to `producers` using `computeIfAbsent`? I'm wondering if there will be a race condition with multiple threads.